dev/reference/best-practices/pd-scheduling.md

```diff
@@ -0,0 +1,250 @@
+
+- [Leader/Follower/Learner](/dev/glossary.md#leaderfollowerlearner)
+- [Operator](/dev/glossary.md#operator)
+- [Operator Step](/dev/glossary.md#operator-step)
+- [Pending/Down](/dev/glossary.md#pendingdown)
+- [Region/Peer/Raft Group](/dev/glossary.md#regionpeerraft-group)
+- [Region Split](/dev/glossary.md#region-split)
+- [Scheduler](/dev/glossary.md#scheduler)
+- [Store](/dev/glossary.md#store)
+
+> **Note:**
+>
+> This document initially targets TiDB 3.0. Though some functions are not supported in earlier versions (2.x), but this document still can be used as a reference for similar principles.
+
+## PD scheduling policies
+
+This section introduces the principle and process involved in the scheduling system.
+
+### Scheduling process
+
+Scheduling process generally has three steps:
+
+1. Collect information
+
+    Each TiKV node periodically reports two types of heartbeats to PD: `StoreHeartbeat` and `RegionHeartbeat`. `StoreHeartbeat` contains the overall information of Stores including disk capacity, remaining storage, read/write traffic. `RegionHeartbeat` contains the overall information of Regions including the range of each Region, peer distribution, peer status, data volume, and reads and writes traffic. PD collects and restores these information for scheduling decisions.
+
+2. Generate Operators
+
+    Different schedulers generate Operators to be executed based on their own logic and requirements, with the consideration of constraints and limitations include:
+
+     - Do not add Peers to a Store with abnormal states (disconnect, down, busy, out of space, with extensive sent/received Snapshots)
+     - Do not balance Regions with abnormal states
+     - Do not attempt to transfer a Leader to a Pending Peer
+     - Do not attempt to remove a Leader directly
+     - Do not break the physical isolation of various Region Peers
+     - Do not break constraints such as Label property
+
+3. Execute scheduling tasks
+
+    The generated Operator first joins a queue managed by `OperatorController` rather than be executed immediately. The `OperatorController` takes the Operator out of the queue and executes it with a certain amount of concurrency based on the configuration. The procedure is to distribute each Operator Step to the corresponding Region Leader. In the end, the Operator is marked as finish or timeout and removed from the execution list.
+
+### Load balancing
+
+Region primarily relies on `balance-leader` and `balance-region` schedulers to achieve load balance. Both schedulers target distributing Regions evenly across all Stores in the cluster but with separate focuses: `balance-leader` attends Region Leader to balance incoming client requests. `balance-region` centers each Region Peer to redistribute the pressure of storage while avoid exceptions like out of storage space.
```

Comment:
I think the 'attends' in this sounds a bit awkward:

>  `balance-leader` **attends** Region Leader to balance incoming client requests.

Similarly 'centers each' also sounds quite strange in:

>  `balance-region` **centers each** Region Peer to redistribute the pressure of storage while avoid exceptions like out of storage space.

Maybe there is a different way we can work them to make the meaning more clear?

Reply:
Thank you, Ana. The intention here is to say:
"`balance-leader` focuses on Region Leaders while `balance-region` focuses on each Region Peer..." But I don't want to overuse "focus" in one paragraph.
How about "deal with"?
"`balance-leader` deals with Region Leaders to balance incoming client requests, whereas `balance-region` deals with each Region Peer to redistribute the pressure of storage and avoid exceptions like out of storage space."

Reply:
I think this is a great improvement!

Reply:
If you want to avoid repeating "Deal with" you could do like:

> `balance-leader` deals with Region Leaders to balance incoming client requests, whereas `balance-region` concerns itself with each Region Peer to redistribute the pressure of storage and avoid exceptions like out of storage space.